app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,40 @@
+require 'bcrypt'
+class PasswordResetsController < ApplicationController
+  def new
+    @page_title = "Reset Password | Haxxor News"
+  end
+  
+  def create
+    @user = User.find_by(email: params[:email])
+    if @user
+      flash[:notice] = "We've sent a reset password link to the given email."
+      PasswordResetMailer.with(user: @user, token: @user.password_token).reset_email.deliver_now
```

Comment:
I think you can use `deliver_later` so the controller doesn't wait for the mail to actually send before responding. ActionMailer will still deliver it, but probably a second or two later in the background.

Reply:
By default, this is done inline, right? So not really in the background?

Reply:
I don't remember what it really does in each environment, but generally not a good idea to use `deliver_now` because code execution waits on it. `deliver_later` will put it into an ActiveJob queue that'll _probably_ get processed immediately, but at least it's no longer blocking the code execution. I think in the test environment `deliver_later` is processed like `perform_now`; i don't remember what it does in dev.

Reply:
I agree that the use of `deliver_later` makes this a bit more future-proof when a back-end other than `:inline` is used for Active Job. I just don't think `inline` does anything in the background by default, so I don't want that misconception floating out there.

Reply:
Note: Switched to deliver_later